v0/destinations/mautic/utils.js

```diff
@@ -0,0 +1,238 @@
+const { set } = require("lodash");
+const { CustomError, getFieldValueFromMessage } = require("../../util");
+const { BASE_URL } = require("./config");
+const { httpGET } = require("../../../adapters/network");
+const {
+  processAxiosResponse
+} = require("../../../adapters/utils/networkUtils");
+
+// check if input is a date or not
+const isDate = date => {
+  // eslint-disable-next-line no-restricted-globals
+  return new Date(date) !== "Invalid Date" && !isNaN(new Date(date));
+};
+
+// Map for Mapping the Rudder Event Fields to Mautic Event Fields (that can be used for lookup )
+const lookupFieldMap = {
+  title: "title",
+  firstName: "firstname",
+  lastName: "lastname",
+  role: "role",
+  phone: "phone",
+  city: "city",
+  email: "email",
+  state: "state",
+  zipcode: "zipcode",
+  country: "country"
+};
+
+/** @param {*} subDomainName
+ * @param {*} propertyName
+ * @param {*} value
+ * @returns  Axios call Url
+ * creates the axios url using the subDomainName for basic url
+ * and propertyName and Value for filters
+ */
+function createAxiosUrl(subDomainName, propertyName, value) {
+  return `${BASE_URL.replace(
+    "subDomainName",
+    subDomainName
+  )}/contacts?where%5B0%5D%5Bcol%5D=${propertyName}&where%5B0%5D%5Bexpr%5D=eq&where%5B0%5D%5Bval%5D=${value}`;
+}
+/**
+ * @param {*} inputText
+ * @returns Boolean Value
+ * for validating email match function outdated
+ */
+function validateEmail(inputText) {
+  const mailformat = /^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,3}$/;
+  return mailformat.test(inputText);
+}
+
+/**
+ * @param {*} inputText
+ * @returns Boolean Value
+ * for validating Phone match function outdated
+ * */
+function validatePhone(inputText) {
+  const phoneno = /^\d{10}$/;
+  return phoneno.test(inputText);
+}
+
+
+/**
+ * @param {*} message
+ * @returns address! and address2 
+ * Constructs the address1 and address2 field
+ * if address is given as string or object
+ * */
```

Comment:
Need to refactor this code also allow user to provide 2 address lines inside address field